From e1142bbc7ea4d9df54627cd56b95906bb18fa0d6 Mon Sep 17 00:00:00 2001
From: Luciana Fujii Pontello <lucianafujii@collabora.co.uk>
Date: Wed, 19 Oct 2011 15:26:40 -0400
Subject: [PATCH 109/154] pvrvideosink: Add config arg to use external headers

Add --with-pvr-external-dependencies argument to use system installed
pvr headers.
---
 configure.ac                |    8 +++++++-
 sys/pvr2d/gstpvrvideosink.c |    2 ++
 2 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index cd36a23..6ce51f3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -229,6 +229,8 @@ if test "x$BUILD_EXAMPLES" = "xyes"; then
 fi
 AM_CONDITIONAL(HAVE_GTK, test "x$HAVE_GTK" = "xyes")
 
+
+
 dnl Needed for GtkBuilder to autoconnect signals
 PKG_CHECK_MODULES(GMODULE_EXPORT, gmodule-export-2.0, HAVE_GMODULE_EXPORT=yes, HAVE_GMODULE_EXPORT=no)
 
@@ -1308,7 +1310,11 @@ dnl *** PVR ***
  PKG_CHECK_MODULES([PVR], [libtimemmgr])
  AC_SUBST(PVR_CFLAGS)
  AC_SUBST(PVR_LIBS)
-
+ AC_ARG_WITH([pvr-external-headers],
+              AC_HELP_STRING([--with-pvr-external-headers], [Use system installed PVR2D headers]),
+              [AS_IF([test "x$with_pvr_external_headers" = "xno"],
+                     [PVR_CFLAGS="$PVR_CFLAGS -I\$(srcdir)/pvr_includes"])],
+              [PVR_CFLAGS="$PVR_CFLAGS -I\$(srcdir)/pvr_includes"])
 
 dnl *** rsvg ***
 translit(dnm, m, l) AM_CONDITIONAL(USE_RSVG, true)
diff --git a/sys/pvr2d/gstpvrvideosink.c b/sys/pvr2d/gstpvrvideosink.c
index bca1e3b..aeabd41 100644
--- a/sys/pvr2d/gstpvrvideosink.c
+++ b/sys/pvr2d/gstpvrvideosink.c
@@ -32,6 +32,8 @@
 
 #define LINUX
 #include "services.h"
+#include "img_defs.h"
+#include "servicesext.h"
 
 #define DEFAULT_QUEUE_SIZE 12
 #define DEFAULT_MIN_QUEUED_BUFS 1
-- 
1.7.5.4

